services-core/src/main/java/com/mapbox/core/utils/TextUtils.java

```diff
@@ -111,27 +126,44 @@ public static String formatRadiuses(double[] radiuses) {
     return join(";", radiusesFormatted);
   }
 
+  public static String formatRadiuses(List<Double> radiuses) {
+    if (radiuses == null || radiuses.size() == 0) {
+      return null;
+    }
+
+    String[] radiusesFormatted = new String[radiuses.size()];
+    for (int i = 0; i < radiuses.size(); i++) {
+      if (radiuses.get(i) == Double.POSITIVE_INFINITY) {
+        radiusesFormatted[i] = "unlimited";
+      } else {
+        radiusesFormatted[i] = String.format(Locale.US, "%s",
+                TextUtils.formatCoordinate(radiuses.get(i)));
+      }
+    }
+    return join(";", radiusesFormatted);
+  }
+
   /**
    * Formats the bearing variables from the raw values to a string which can than be used for the
    * request URL.
    *
-   * @param bearings a List of doubles representing bearing values
+   * @param bearings a List of list of doubles representing bearing values
    * @return a string with the bearing values
    * @since 3.0.0
    */
-  public static String formatBearing(List<Double[]> bearings) {
+  public static String formatBearing(List<List<Double>> bearings) {
     if (bearings.isEmpty()) {
       return null;
     }
 
     String[] bearingFormatted = new String[bearings.size()];
     for (int i = 0; i < bearings.size(); i++) {
-      if (bearings.get(i).length == 0) {
+      if (bearings.get(i).size() == 0) {
         bearingFormatted[i] = "";
       } else {
         bearingFormatted[i] = String.format(Locale.US, "%s,%s",
-          TextUtils.formatCoordinate(bearings.get(i)[0]),
-          TextUtils.formatCoordinate(bearings.get(i)[1]));
+                TextUtils.formatCoordinate(bearings.get(i).get(0)),
```

Comment:
NIT formatting (actually `checkstyle` is ❌ in CI 😅 https://circleci.com/gh/mapbox/mapbox-java/3900?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link)

For Java code we use [this](https://github.com/mapbox/mapbox-navigation-android/tree/master/config/checkstyle) checkstyle and code style. You can find instructions on how to set both in [Setting up Mapbox checkstyle](https://github.com/mapbox/mapbox-gl-native-android/wiki/Setting-up-Mapbox-checkstyle).